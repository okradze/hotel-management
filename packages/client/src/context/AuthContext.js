import React from 'react'

const Context = React.createContext()

const AuthContext = ({ children }) => {
    const [user, setUser] = React.useState()

    return (
        <Context.Provider value={{ user, setUser }}>
            {children}
        </Context.Provider>
    )
}

export { AuthContext as default, Context }
