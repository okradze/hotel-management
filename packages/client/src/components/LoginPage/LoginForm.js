import React, { useState, useContext } from 'react'
import { useMutation } from '@apollo/react-hooks'
import { withRouter } from 'react-router-dom'
import { Context } from '../../context/AuthContext'
import Loader from '../Loader/Loader'
import { LOGIN } from './gql'
import Input from '../Input'

const LoginForm = ({ history, setIsLogin }) => {
    const [email, setEmail] = useState('')
    const [password, setPassword] = useState('')

    const { setIsAuth } = useContext(Context)
    const [login, { error, loading }] = useMutation(LOGIN, {
        variables: {
            email,
            password,
        },
        onCompleted: handleCompleted,
    })

    function handleEmailChange(e) {
        setEmail(e.target.value)
    }

    function handlePasswordChange(e) {
        setPassword(e.target.value)
    }

    function handleSubmit(e) {
        e.preventDefault()
        if (
            email.length >= 1 &&
            email.length <= 50 &&
            password.length >= 1 &&
            password.length <= 50
        ) {
            login()
        }
    }

    function handleCompleted(data) {
        if (data.login === true) {
            setIsAuth(true)
            history.push('/dashboard')
        }
    }

    return (
        <div className="LoginPage-wrapper">
            <div className="LoginPage-form">
                <form onSubmit={e => handleSubmit(e)} className="LoginForm">
                    <Input
                        text="ელ-ფოსტა:"
                        value={email}
                        onChange={handleEmailChange}
                        type="text"
                        id="email"
                    />
                    <Input
                        text="პაროლი:"
                        value={password}
                        onChange={handlePasswordChange}
                        type="password"
                        id="password"
                    />

                    {error && (
                        <p className="LoginForm__error error">
                            მომხმარებელი არ მოიძებნა
                        </p>
                    )}
                    <button type="submit" className="button button--secondary">
                        {loading ? (
                            <>
                                <Loader />
                            </>
                        ) : (
                            'შესვლა'
                        )}
                    </button>
                </form>
                <div>
                    <p
                        role="link"
                        tabIndex="0"
                        onKeyPress={e => {
                            if (e.which === 13) {
                                setIsLogin(false)
                            }
                        }}
                        onClick={() => setIsLogin(false)}
                        className="form-link"
                    >
                        რეგისტრაცია
                    </p>
                </div>
            </div>
        </div>
    )
}

export default withRouter(LoginForm)
