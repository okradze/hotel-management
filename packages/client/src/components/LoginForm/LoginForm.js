import React from 'react'
import { useMutation } from '@apollo/react-hooks'
import { withRouter } from 'react-router-dom'
import { Context } from '../../context/AuthContext'
import { LOGIN } from '../LoginPage/gql'
import Input from '../Input/Input'
import withLoader from '../Loader'

export const LoginForm = ({ history, setIsLogin }) => {
    const [email, setEmail] = React.useState('')
    const [password, setPassword] = React.useState('')
    const [error, setError] = React.useState('')
    const { setUser } = React.useContext(Context)

    const [login, { loading }] = useMutation(LOGIN, {
        variables: {
            email,
            password,
        },
        onCompleted: handleCompleted,
        onError: err => {
            if (err.message.endsWith('no_user')) {
                setError('მომხმარებელი არ მოიძებნა')
            }
        },
    })

    function handleEmailChange(e) {
        setEmail(e.target.value)
    }

    function handlePasswordChange(e) {
        setPassword(e.target.value)
    }

    function handleSubmit(e) {
        e.preventDefault()
        if (
            email.length >= 1 &&
            email.length <= 100 &&
            password.length >= 1 &&
            password.length <= 64
        ) {
            login()
        }
    }

    function handleCompleted({ login }) {
        if (login) {
            setUser({ createdAt: login.createdAt })
            history.push('/dashboard')
        }
    }

    const ButtonWithLoader = withLoader(() => 'შესვლა')

    return (
        <div className="LoginPage-wrapper">
            <div className="LoginPage-form">
                <form onSubmit={e => handleSubmit(e)} className="LoginForm">
                    <Input
                        text="ელ-ფოსტა:"
                        value={email}
                        onChange={handleEmailChange}
                        type="text"
                        id="email"
                    />
                    <Input
                        text="პაროლი:"
                        value={password}
                        onChange={handlePasswordChange}
                        type="password"
                        id="password"
                    />

                    {error && <p className="LoginForm__error error">{error}</p>}

                    <button type="submit" className="button button--secondary">
                        <ButtonWithLoader isLoading={loading} />
                    </button>
                </form>

                <button
                    type="button"
                    onKeyPress={e => {
                        if (e.which === 13) {
                            setIsLogin(false)
                        }
                    }}
                    onClick={() => setIsLogin(false)}
                    className="button--transparent form-link"
                >
                    რეგისტრაცია
                </button>
            </div>
        </div>
    )
}

export default withRouter(LoginForm)
