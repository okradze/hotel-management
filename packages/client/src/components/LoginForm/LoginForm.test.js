import { LoginForm } from './LoginForm'
import { Context } from '../../context/AuthContext'

let history
let setIsLogin
let wrapper

beforeEach(() => {
    history = {
        push: jest.fn(),
    }
    setIsLogin = jest.fn()
    wrapper = shallow(
        <Context.Provider
            value={{
                setIsUser: jest.fn(),
            }}
        >
            <LoginForm history={history} setIsLogin={setIsLogin} />
        </Context.Provider>,
    )
})

describe('LoginForm', () => {
    it('renders', () => {
        expect(wrapper).toMatchSnapshot()
    })
})
