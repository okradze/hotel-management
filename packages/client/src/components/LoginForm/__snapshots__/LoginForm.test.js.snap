// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders 1`] = `
<Fragment>
  <form
    className="LoginForm"
    onSubmit={[Function]}
  >
    <Input
      id="email"
      onChange={[Function]}
      placeholder="შეიყვანეთ ელ-ფოსტა"
      text="ელ-ფოსტა"
      type="text"
      value="abc"
    />
    <Input
      id="password"
      onChange={[Function]}
      placeholder="შეიყვანეთ პაროლი"
      text="პაროლი"
      type="password"
      value="abc"
    />
    <p
      className="LoginForm__error error"
    >
      abc
    </p>
    <button
      className="button button--secondary"
      type="submit"
    >
      <Component
        isLoading={false}
      />
    </button>
  </form>
  <button
    className="button--transparent form-link"
    onClick={[Function]}
    onKeyPress={[Function]}
    type="button"
  >
    რეგისტრაცია →
  </button>
</Fragment>
`;
